File: RepoTransBench/source_projects/Java/NetEase_arrow/src/test/java/com/netease/qa/ConvertPublicTest.java
package com.netease.qa;

import org.junit.*;
import java.io.*;
import org.apache.commons.io.FileUtils;
import static org.junit.Assert.*;

public class ConvertPublicTest {

    private File gbkFile;
    private File utf8File;

    @Before
    public void setUp() throws Exception {
        gbkFile = File.createTempFile("public_testfile", ".txt");
        utf8File = File.createTempFile("public_utf8file", ".txt");
        // Write different Chinese characters "世界" (means "world") in GBK encoding
        FileOutputStream fos = new FileOutputStream(gbkFile);
        fos.write("世界".getBytes("GBK"));
        fos.close();
        if (utf8File.exists()) utf8File.delete();
    }

    @After
    public void tearDown() throws Exception {
        if (gbkFile.exists()) gbkFile.delete();
        if (utf8File.exists()) utf8File.delete();
        File tmp = new File(gbkFile.getAbsolutePath() + ".tmp");
        if (tmp.exists()) tmp.delete();
    }

    @Test
    public void testConvertGbkToUtf8InPlace() throws Exception {
        String[] args = {gbkFile.getAbsolutePath()};
        Convert.main(args);
        // File should remain (be replaced). Read as UTF-8.
        String content = FileUtils.readFileToString(gbkFile, "UTF-8");
        assertTrue(content.contains("世界"));
    }

    @Test
    public void testConvertGbkToUtf8WithDifferentOutputFile() throws Exception {
        String[] args = {gbkFile.getAbsolutePath(), utf8File.getAbsolutePath()};
        Convert.main(args);
        assertTrue(utf8File.exists());
        String content = FileUtils.readFileToString(utf8File, "UTF-8");
        assertTrue(content.contains("世界"));
    }

    @Test
    public void testIOExceptionIsHandled() throws Exception {
        // Use a definitely non-existent file to trigger IOException (different path)
        String[] args = {"/definitely/doesnotexist/inputfile_public.txt"};
        // Should not throw, just print stack trace
        Convert.main(args);
    }
}

File: RepoTransBench/target_projects/Java/Go/NetEase_arrow/public_tests/convert_public_test.go
package public_tests

import (
	"os"
	"testing"

	"github.com/stretchr/testify/assert"
	"netease_arrow/tests"
)

// Simulates ConvertPublicTest with different test string ("世界": "world")
func TestConvertGbkToUtf8InPlace(t *testing.T) {
	gbkFile, err := os.CreateTemp("", "public_testfile*.txt")
	assert.NoError(t, err)
	defer os.Remove(gbkFile.Name())

	utf8File, err := os.CreateTemp("", "public_utf8file*.txt")
	assert.NoError(t, err)
	defer os.Remove(utf8File.Name())

	// Write Chinese "世界" in GBK encoding
	err = tests.WriteGBKFile(gbkFile.Name(), "世界")
	assert.NoError(t, err)
	tests.RemoveIfExists(utf8File.Name(), t)

	// Simulate Convert.main([gbkFile])
	err = tests.ConvertFileInPlaceGbkToUtf8(gbkFile.Name())
	assert.NoError(t, err)
	content, err := tests.ReadUTF8File(gbkFile.Name())
	assert.NoError(t, err)
	assert.Contains(t, content, "世界")
}

func TestConvertGbkToUtf8WithDifferentOutputFile(t *testing.T) {
	gbkFile, err := os.CreateTemp("", "public_testfile*.txt")
	assert.NoError(t, err)
	defer os.Remove(gbkFile.Name())

	utf8File, err := os.CreateTemp("", "public_utf8file*.txt")
	assert.NoError(t, err)
	defer os.Remove(utf8File.Name())

	// Write Chinese "世界" in GBK encoding
	err = tests.WriteGBKFile(gbkFile.Name(), "世界")
	assert.NoError(t, err)
	tests.RemoveIfExists(utf8File.Name(), t)

	// Convert (simulate Convert.main([gbkFile, utf8File]))
	err = tests.ConvertFileGbkToUtf8Out(gbkFile.Name(), utf8File.Name())
	assert.NoError(t, err)

	// utf8File should exist and contain "世界"
	_, err = os.Stat(utf8File.Name())
	assert.NoError(t, err)

	content, err := tests.ReadUTF8File(utf8File.Name())
	assert.NoError(t, err)
	assert.Contains(t, content, "世界")
}

func TestIOExceptionIsHandled(t *testing.T) {
	nonExistent := "/definitely/doesnotexist/inputfile_public.txt"
	err := tests.ConvertFileInPlaceGbkToUtf8(nonExistent)
	assert.Error(t, err)
}